crates/nu-cmd-extra/src/extra/bits/mod.rs

```diff
@@ -95,7 +100,68 @@ fn get_input_num_type(val: i64, signed: bool, number_size: NumberBytes) -> Input
                     InputNumType::Eight
                 }
             }
-            NumberBytes::Invalid => InputNumType::Four,
         }
     }
 }
+
+fn binary_op<F>(lhs: &Value, rhs: &Value, little_endian: bool, f: F) -> Value
+where
+    F: Fn((i64, i64)) -> i64,
+{
+    let span = lhs.span();
+    match (lhs, rhs) {
+        (Value::Int { val: lhs, .. }, Value::Int { val: rhs, .. }) => {
+            Value::int(f((*lhs, *rhs)), span)
+        }
+        (Value::Binary { val: lhs, .. }, Value::Binary { val: rhs, .. }) => {
+            let (lhs, rhs, max_len, min_len) = match (lhs.len(), rhs.len()) {
+                (max, min) if max > min => (lhs, rhs, max, min),
+                (min, max) => (rhs, lhs, max, min),
+            };
+
+            let pad = iter::repeat(0).take(max_len - min_len);
+
+            let mut a;
+            let mut b;
+
+            let padded: &mut dyn Iterator<Item = u8> = if little_endian {
+                a = rhs.iter().copied().chain(pad);
+                &mut a
+            } else {
+                b = pad.chain(rhs.iter().copied());
+                &mut b
+            };
+
+            let bytes: Vec<u8> = lhs
+                .iter()
+                .copied()
+                .zip(padded)
+                .map(|(lhs, rhs)| f((lhs as i64, rhs as i64)) as u8)
+                .collect();
+
+            Value::binary(bytes, span)
+        }
+        (Value::Binary { .. }, Value::Int { .. }) | (Value::Int { .. }, Value::Binary { .. }) => {
+            Value::error(
+                ShellError::PipelineMismatch {
+                    exp_input_type: "input, and argument, to be both int or both binary"
+                        .to_string(),
+                    dst_span: rhs.span(),
+                    src_span: span,
+                },
+                span,
+            )
+        }
+        // Propagate errors by explicitly matching them before the final case.
+        (e @ Value::Error { .. }, _) | (_, e @ Value::Error { .. }) => e.clone(),
+        (other, Value::Int { .. } | Value::Binary { .. }) | (_, other) => Value::error(
+            ShellError::OnlySupportsThisInputType {
+                exp_input_type: "int or binary".into(),
+                wrong_type: other.get_type().to_string(),
+                dst_span: other.span(),
+                src_span: span,
```

Comment:
One of these spans needs to be `call.head`, I think. Here's the old error:
![image](https://github.com/nushell/nushell/assets/45539777/337c82bb-822a-4b0d-bd7f-fef412590705)

and the new one:

![image](https://github.com/nushell/nushell/assets/45539777/46dbfcad-89e9-4355-9891-695810281df7)

Reply:
should be fixed now, spans are confusing
thanks again!